### PR TITLE
[model, config, docs, ci] feat: support dense Qwen3.5 ChunkMBS

### PR DESCRIPTION
**Pull request by WenZheWang**
### What does this PR do?

> Extend the packed ChunkMBS support introduced in #898 to dense Qwen3.5 packed SFT. The feature remains disabled by default and reuses the shared decoder-layer ChunkMBS implementation without changing generated model code or shared runtime behavior.

### Checklist Before Starting

- Search for relative PRs/issues and link here: #898 is the merged implementation baseline; no overlapping open Qwen3.5 ChunkMBS PR was found.
- PR title follows `[{modules}] {type}: {description}` format.

### Test

- `make quality`: passed (`ruff check` and `ruff format --check`, 594 files).
- `pytest -q -rs tests/parallel/test_chunk_mbs.py`: 47 passed, 1 skipped. The skipped case is the existing real FSDP2 test because macOS does not support CPU FSDP2 process groups.
- Parsed `configs/multimodal/qwen3_5/qwen3_5_vl.yaml` through `VeOmniVLMArguments`: `chunk_mbs_config.enable=false` and `chunk_mbs=2`.
- Real FlashAttention, FLA, and NPU fused-kernel execution was not run locally and still requires accelerator validation.

### API and Usage Example

```yaml
train:
  chunk_mbs_config:
    enable: true
    chunk_mbs: 2
```

`chunk_mbs` is the number of packed samples per decoder-layer chunk. Dense Qwen3.5 supports this with FSDP2 and non-reentrant gradient checkpointing. Qwen3.5-MoE, SP, TP/PP, `torch.compile`, `pad_to_length`, DPO, and RL remain unsupported.

### Design & Code Changes

> - Expose disabled-by-default ChunkMBS settings in the dense Qwen3.5 multimodal example config.
> - Add model-level tests using the real Qwen3.5 decoder layer for full-attention and linear-attention routing.
> - Verify forward output, input gradients, parameter gradients, exact per-chunk cumulative-length metadata, and non-reentrant checkpoint recomputation order.
> - Model GatedDeltaNet segment-state resets with a stateful CPU test double; keep accelerator-specific kernel validation explicitly out of the CPU test claim.
> - Document the Qwen3.5 support boundary and clarify that only ChunkMBS cut boundaries must align across full- and linear-attention metadata.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [x] Applied pre-commit checks
- [x] Added/updated documentation
- [x] Added model-level tests; hardware fused-kernel validation is documented as not run locally

### CODE REVIEW

**Comment by WenzheWang** [comment]
@Luosuu All checks are green for current head `6f6843bc`, the PR is mergeable, and there are no unresolved review threads.

Current-head local validation also passes: `make quality` and `pytest -q -rs tests/parallel/test_chunk_mbs.py` (`47 passed, 1 skipped`; the existing real FSDP2 case is skipped because macOS does not support CPU FSDP2 process groups).

Upstream `main` is now at `a6cdb3b1`. I inspected the intervening #918, #919, and #922 changes; they cover the explicitly unsupported tail-padding path, NPU HF checkpoint saving, and MoE Quack GEMM, so they do not overlap the supported dense Qwen3.5 ChunkMBS path in this PR.

Could you please review and merge when convenient? Thanks.

**Review by Luosuu — APPROVED** [review]
